assets/app/scripts/controllers/create.js

```diff
@@ -94,6 +107,213 @@ angular.module('openshiftConsole')
       $scope.filter.tag = tag;
     };
 
+    $scope.aceLoaded = function(editor) {
+      $scope.editor.session = editor.getSession();
+      $scope.editor.session.setOption('tabSize', 2);
+      $scope.editor.session.setOption('useSoftTabs', true);
+      editor.setDragDelay = 0;
+      editor.$blockScrolling = Infinity;
+
+      $('.from-file .editor').animate({
+        height: Math.floor(window.innerHeight * 0.50) + 'px'
+      }, 30, function() {
+        editor.resize();
+      });
+    };
+
+    // Check if the editor isn't empty to disable the 'Add' button. Also check in what format the input is in (JSON/YAML) and change
+    // the editor accordingly.
+    $scope.aceChanged = function() {
+      $scope.editor.empty = false;
+      if (!$scope.editor.content) {
+        $scope.editor.empty = true;
+      }
+      try {
+        JSON.parse($scope.editor.content);
+        $scope.editor.session.setMode("ace/mode/json");
+      } catch (e) {
+        try {
+          YAML.parse($scope.editor.content);
+          $scope.editor.session.setMode("ace/mode/yaml");
+        } catch (e) {
+          return;
+        }
+      }
```

Comment:
We should debounce these checks so we're not parsing and reparsing on every keystroke.

https://lodash.com/docs#debounce